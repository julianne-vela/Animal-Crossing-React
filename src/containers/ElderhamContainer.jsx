import React, { Component } from 'react';
import VillagerList from '../components/villagers/VillagerList';
import { findVillagers, searchBySpecies } from '../services/acnhApi';
import LoadScreen from '../components/app/LoadScreen';
import Controls from '../components/search/Controls';
import style from '../components/styles.css';

export default class ElderhamContainer extends Component {
  state = {
    loading: false,
    villagers: [],
    searchQuery: '',
  };

  async componentDidMount() {
    this.setState({ loading: true });
    const villagers = await findVillagers();

    this.setState({
      villagers,
      loading: false,
    });
  }

  handleQueryChange = ({ target }) => {
    this.setState({ searchQuery: target.value });
  };

  handleSubmit = async (e) => {
    e.preventDefault();
    this.setState({ loading: true });

    const { searchQuery } = this.state;

    const villagers = await findVillagers();
    this.setState({ villagers });

    const villagerList = await searchBySpecies(searchQuery, villagers);

    this.setState({
      villagers: villagerList,
      loading: false,
      searchQuery: '',
    });
  };

  render() {
    const { loading, villagers, searchQuery } = this.state;

    if (loading) return <LoadScreen />;

    return (
      <main className={style.mainContainer}>
        <Controls
          onSubmit={this.handleSubmit}
          //   species={searchQuery}
          onInputChange={this.handleQueryChange}
        />
        <VillagerList villagers={villagers} />
      </main>
    );
  }
}
