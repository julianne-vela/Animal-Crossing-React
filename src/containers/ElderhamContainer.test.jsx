import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import ElderhamContainer from './ElderhamContainer';
import { MemoryRouter } from 'react-router';

// snapshot test all presentational components
// behavior test - on changing search refetch articles

describe('Elderham Container Tests', () => {
  it('should render a list of 25 villagers to the page', async () => {
    render(
      <MemoryRouter>
        <ElderhamContainer />
      </MemoryRouter>
    );

    await screen.findAllByRole('img');

    const ul = await screen.findByRole('list', {
      name: 'villagers',
    });
    expect(ul).toMatchSnapshot();
  });

  it('should render search results', async () => {
    render(
      <MemoryRouter>
        <ElderhamContainer />
      </MemoryRouter>
    );

    screen.findAllByRole('img');

    const input = await screen.findByLabelText('Search by Species');
    const button = await screen.findByRole('button', { name: 'search-submit' });
    const form = await screen.findByRole('form', { name: 'search-query' });

    expect(form).toMatchSnapshot();
    expect(button).toMatchSnapshot();

    await userEvent.type(input, 'pig');
    await userEvent.click(button);

    return waitFor(async () => {
      const searchResults = await screen.findByRole('list', {
        name: 'villagers',
      });
      expect(searchResults.childElementCount).toBe(1);
    });
  });
});
