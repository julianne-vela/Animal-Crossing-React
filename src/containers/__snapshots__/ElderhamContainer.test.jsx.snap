// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Elderham Container Tests should render a list of 25 villagers to the page 1`] = `
<ul
  aria-label="villagers"
  class="villagersList"
>
  <li
    class="cardBox"
  >
    <a
      href="/villagers/5f5fb4bbbfd05c2aed82e460"
    >
      <article
        class="villagerCard"
      >
        <p>
          Admiral
        </p>
        <img
          alt="Admiral"
          src="https://vignette.wikia.nocookie.net/animalcrossing/images/e/ed/Admiral_NH.png/revision/latest?cb=20200802081138"
        />
        <p>
          bird
        </p>
      </article>
    </a>
  </li>
  <li
    class="cardBox"
  >
    <a
      href="/villagers/5f5fb4bbbfd05c2aed82e461"
    >
      <article
        class="villagerCard"
      >
        <p>
          Agent S
        </p>
        <img
          alt="Agent S"
          src="https://vignette.wikia.nocookie.net/animalcrossing/images/a/a7/Agent_S_NH.png/revision/latest?cb=20200719092526"
        />
        <p>
          squirrel
        </p>
      </article>
    </a>
  </li>
  <li
    class="cardBox"
  >
    <a
      href="/villagers/5f5fb4bbbfd05c2aed82e462"
    >
      <article
        class="villagerCard"
      >
        <p>
          Agnes
        </p>
        <img
          alt="Agnes"
          src="https://vignette.wikia.nocookie.net/animalcrossing/images/4/42/Agnes_NH.png/revision/latest?cb=20200804230432"
        />
        <p>
          pig
        </p>
      </article>
    </a>
  </li>
  <li
    class="cardBox"
  >
    <a
      href="/villagers/5f5fb4bbbfd05c2aed82e463"
    >
      <article
        class="villagerCard"
      >
        <p>
          Al
        </p>
        <img
          alt="Al"
          src="https://vignette.wikia.nocookie.net/animalcrossing/images/c/c4/Al_NH.png/revision/latest?cb=20200802150354"
        />
        <p>
          gorilla
        </p>
      </article>
    </a>
  </li>
  <li
    class="cardBox"
  >
    <a
      href="/villagers/5f5fb4bbbfd05c2aed82e464"
    >
      <article
        class="villagerCard"
      >
        <p>
          Alfonso
        </p>
        <img
          alt="Alfonso"
          src="https://vignette.wikia.nocookie.net/animalcrossing/images/e/e4/Alfonso_NH.png/revision/latest?cb=20200817164717"
        />
        <p>
          alligator
        </p>
      </article>
    </a>
  </li>
  <li
    class="cardBox"
  >
    <a
      href="/villagers/5f5fb4bbbfd05c2aed82e465"
    >
      <article
        class="villagerCard"
      >
        <p>
          Alice
        </p>
        <img
          alt="Alice"
          src="https://vignette.wikia.nocookie.net/animalcrossing/images/e/e1/Alice_NH.png/revision/latest?cb=20200803181400"
        />
        <p>
          koala
        </p>
      </article>
    </a>
  </li>
  <li
    class="cardBox"
  >
    <a
      href="/villagers/5f5fb4bbbfd05c2aed82e46c"
    >
      <article
        class="villagerCard"
      >
        <p>
          Ankha
        </p>
        <img
          alt="Ankha"
          src="https://vignette.wikia.nocookie.net/animalcrossing/images/5/56/Ankha_NH.png/revision/latest?cb=20200817172044"
        />
        <p>
          cat
        </p>
      </article>
    </a>
  </li>
  <li
    class="cardBox"
  >
    <a
      href="/villagers/5f5fb4bbbfd05c2aed82e46a"
    >
      <article
        class="villagerCard"
      >
        <p>
          Angus
        </p>
        <img
          alt="Angus"
          src="https://vignette.wikia.nocookie.net/animalcrossing/images/5/5f/Angus_NH.png/revision/latest?cb=20200802083758"
        />
        <p>
          bull
        </p>
      </article>
    </a>
  </li>
  <li
    class="cardBox"
  >
    <a
      href="/villagers/5f5fb4bbbfd05c2aed82e466"
    >
      <article
        class="villagerCard"
      >
        <p>
          Alli
        </p>
        <img
          alt="Alli"
          src="https://vignette.wikia.nocookie.net/animalcrossing/images/8/8e/Alli_NH.png/revision/latest?cb=20200802061731"
        />
        <p>
          alligator
        </p>
      </article>
    </a>
  </li>
  <li
    class="cardBox"
  >
    <a
      href="/villagers/5f5fb4bbbfd05c2aed82e46b"
    >
      <article
        class="villagerCard"
      >
        <p>
          Anicotti
        </p>
        <img
          alt="Anicotti"
          src="https://vignette.wikia.nocookie.net/animalcrossing/images/8/80/Anicotti_NH.png/revision/latest?cb=20200802164003"
        />
        <p>
          mouse
        </p>
      </article>
    </a>
  </li>
  <li
    class="cardBox"
  >
    <a
      href="/villagers/5f5fb4bbbfd05c2aed82e467"
    >
      <article
        class="villagerCard"
      >
        <p>
          Amelia
        </p>
        <img
          alt="Amelia"
          src="https://vignette.wikia.nocookie.net/animalcrossing/images/a/ab/Amelia_NH.png/revision/latest?cb=20200817171545"
        />
        <p>
          eagle
        </p>
      </article>
    </a>
  </li>
  <li
    class="cardBox"
  >
    <a
      href="/villagers/5f5fb4bbbfd05c2aed82e46d"
    >
      <article
        class="villagerCard"
      >
        <p>
          Annalisa
        </p>
        <img
          alt="Annalisa"
          src="https://vignette.wikia.nocookie.net/animalcrossing/images/f/f0/Annalisa_NH.png/revision/latest?cb=20200802063308"
        />
        <p>
          anteater
        </p>
      </article>
    </a>
  </li>
  <li
    class="cardBox"
  >
    <a
      href="/villagers/5f5fb4bbbfd05c2aed82e46f"
    >
      <article
        class="villagerCard"
      >
        <p>
          Antonio
        </p>
        <img
          alt="Antonio"
          src="https://vignette.wikia.nocookie.net/animalcrossing/images/d/d4/Antonio_NH.png/revision/latest?cb=20200802063352"
        />
        <p>
          anteater
        </p>
      </article>
    </a>
  </li>
  <li
    class="cardBox"
  >
    <a
      href="/villagers/5f5fb4bbbfd05c2aed82e46e"
    >
      <article
        class="villagerCard"
      >
        <p>
          Annalise
        </p>
        <img
          alt="Annalise"
          src="https://vignette.wikia.nocookie.net/animalcrossing/images/c/c4/Annalise_NH.png/revision/latest?cb=20200802153943"
        />
        <p>
          horse
        </p>
      </article>
    </a>
  </li>
  <li
    class="cardBox"
  >
    <a
      href="/villagers/5f5fb4bbbfd05c2aed82e468"
    >
      <article
        class="villagerCard"
      >
        <p>
          Anabelle
        </p>
        <img
          alt="Anabelle"
          src="https://vignette.wikia.nocookie.net/animalcrossing/images/d/d8/Anabelle_NH.png/revision/latest?cb=20200802063200"
        />
        <p>
          anteater
        </p>
      </article>
    </a>
  </li>
  <li
    class="cardBox"
  >
    <a
      href="/villagers/5f5fb4bbbfd05c2aed82e470"
    >
      <article
        class="villagerCard"
      >
        <p>
          Apollo
        </p>
        <img
          alt="Apollo"
          src="https://vignette.wikia.nocookie.net/animalcrossing/images/2/28/Apollo_NH.png/revision/latest?cb=20200719092927"
        />
        <p>
          eagle
        </p>
      </article>
    </a>
  </li>
  <li
    class="cardBox"
  >
    <a
      href="/villagers/5f5fb4bbbfd05c2aed82e469"
    >
      <article
        class="villagerCard"
      >
        <p>
          Anchovy
        </p>
        <img
          alt="Anchovy"
          src="https://vignette.wikia.nocookie.net/animalcrossing/images/e/ea/Anchovy_NH.png/revision/latest?cb=20200802081231"
        />
        <p>
          bird
        </p>
      </article>
    </a>
  </li>
  <li
    class="cardBox"
  >
    <a
      href="/villagers/5f5fb4bbbfd05c2aed82e471"
    >
      <article
        class="villagerCard"
      >
        <p>
          Apple
        </p>
        <img
          alt="Apple"
          src="https://vignette.wikia.nocookie.net/animalcrossing/images/5/57/Apple_NH.png/revision/latest?cb=20200802151508"
        />
        <p>
          hamster
        </p>
      </article>
    </a>
  </li>
  <li
    class="cardBox"
  >
    <a
      href="/villagers/5f5fb4bbbfd05c2aed82e472"
    >
      <article
        class="villagerCard"
      >
        <p>
          Astrid
        </p>
        <img
          alt="Astrid"
          src="https://vignette.wikia.nocookie.net/animalcrossing/images/d/d5/Astrid_NH.png/revision/latest?cb=20200802154954"
        />
        <p>
          kangaroo
        </p>
      </article>
    </a>
  </li>
  <li
    class="cardBox"
  >
    <a
      href="/villagers/5f5fb4bbbfd05c2aed82e473"
    >
      <article
        class="villagerCard"
      >
        <p>
          Audie
        </p>
        <img
          alt="Audie"
          src="https://vignette.wikia.nocookie.net/animalcrossing/images/1/1b/Audie_NH.png/revision/latest?cb=20200829012735"
        />
        <p>
          wolf
        </p>
      </article>
    </a>
  </li>
  <li
    class="cardBox"
  >
    <a
      href="/villagers/5f5fb4bbbfd05c2aed82e474"
    >
      <article
        class="villagerCard"
      >
        <p>
          Aurora
        </p>
        <img
          alt="Aurora"
          src="https://vignette.wikia.nocookie.net/animalcrossing/images/5/59/Aurora_NH.png/revision/latest?cb=20200202215217"
        />
        <p>
          penguin
        </p>
      </article>
    </a>
  </li>
  <li
    class="cardBox"
  >
    <a
      href="/villagers/5f5fb4bbbfd05c2aed82e476"
    >
      <article
        class="villagerCard"
      >
        <p>
          Avery
        </p>
        <img
          alt="Avery"
          src="https://vignette.wikia.nocookie.net/animalcrossing/images/d/dd/Avery_NH.png/revision/latest?cb=20200608222354"
        />
        <p>
          eagle
        </p>
      </article>
    </a>
  </li>
  <li
    class="cardBox"
  >
    <a
      href="/villagers/5f5fb4bbbfd05c2aed82e475"
    >
      <article
        class="villagerCard"
      >
        <p>
          Ava
        </p>
        <img
          alt="Ava"
          src="https://vignette.wikia.nocookie.net/animalcrossing/images/c/cc/Ava_NH.png/revision/latest?cb=20200609055741"
        />
        <p>
          chicken
        </p>
      </article>
    </a>
  </li>
  <li
    class="cardBox"
  >
    <a
      href="/villagers/5f5fb4bbbfd05c2aed82e47b"
    >
      <article
        class="villagerCard"
      >
        <p>
          Barold
        </p>
        <img
          alt="Barold"
          src="https://vignette.wikia.nocookie.net/animalcrossing/images/9/9d/Barold_NH.png/revision/latest?cb=20200802141140"
        />
        <p>
          cub
        </p>
      </article>
    </a>
  </li>
  <li
    class="cardBox"
  >
    <a
      href="/villagers/5f5fb4bbbfd05c2aed82e47a"
    >
      <article
        class="villagerCard"
      >
        <p>
          Bangle
        </p>
        <img
          alt="Bangle"
          src="https://vignette.wikia.nocookie.net/animalcrossing/images/6/68/Bangle_NH.png/revision/latest?cb=20200803120044"
        />
        <p>
          tiger
        </p>
      </article>
    </a>
  </li>
</ul>
`;

exports[`Elderham Container Tests should render search results 1`] = `
<form
  aria-label="search-query"
>
  <label
    for="search-query"
  >
    Search by Species
  </label>
  <input
    id="search-query"
    type="search"
  />
  <button
    aria-label="search-submit"
  >
    Search
  </button>
</form>
`;

exports[`Elderham Container Tests should render search results 2`] = `
<button
  aria-label="search-submit"
>
  Search
</button>
`;
